scylla-cql/src/types/serialize/serialize_row.rs

```diff
@@ -0,0 +1,44 @@
+use std::{any::Any, sync::Arc};
+
+use crate::{_macro_internal::ValueList, frame::response::result::ColumnSpec};
+
+pub struct RowSerializationContext<'a> {
+    columns: &'a [ColumnSpec],
+}
+
+impl<'a> RowSerializationContext<'a> {
+    pub fn column_by_idx(&self, i: usize) -> Option<&ColumnSpec> {
+        self.columns.get(i)
+    }
+
+    // TODO: change RowSerializationContext to make this faster
+    pub fn column_by_name(&self, target: &str) -> Option<&ColumnSpec> {
+        self.columns.iter().find(|&c| c.name == target)
+    }
+}
+
+pub trait SerializeRow {
+    fn preliminary_type_check(&self, ctx: &RowSerializationContext<'_>) -> bool;
```

Comment:
I thought that this was supposed to be an associated function, not a method?

Reply:
Ooops, yes - that's what I did in previous version but forgot about in this, sorry